omnisend/includes/SDK/V1/class-product.php

```diff
@@ -0,0 +1,463 @@
+		'created_at',
+		'currency',
+		'default_image_url',
+		'description',
+		'id',
+		'status',
+		'title',
+		'type',
+		'updated_at',
+		'url',
+		'vendor',
+	);
+	private const ARRAY_PROPERTIES    = array(
+		'variants',
+		'images',
+	);
+
+	/**
+	 * @var array $category_ids
+	 */
+	private $category_ids = null;
+
+	/**
+	 * @var string $created_at
+	 */
+	private $created_at = null;
+
+	/**
+	 * @var string $currency
+	 */
+	private $currency = null;
+
+	/**
+	 * @var string $default_image_url
+	 */
+	private $default_image_url = null;
+
+	/**
+	 * @var string $description
+	 */
+	private $description = null;
+
+	/**
+	 * @var string $id
+	 */
+	private $id = null;
+
+	/**
+	 * @var array $images
+	 */
+	private $images = null;
+
+	/**
+	 * @var string $status
+	 */
+	private $status = null;
+
+	/**
+	 * @var string $title
+	 */
+	private $title = null;
+
+	/**
+	 * @var string $type
+	 */
+	private $type = null;
+
+	/**
+	 * @var string $updated_at
+	 */
+	private $updated_at = null;
+
+	/**
+	 * @var string $url
+	 */
+	private $url = null;
+
+	/**
+	 * @var array $variants
+	 */
+	private $variants = null;
+
+	/**
+	 * @var string $vendor
+	 */
+	private $vendor = null;
+
+	/**
+	 * @var array $tags
+	 */
+	private array $tags = array();
+
+	/**
+	 * Validates product properties
+	 *
+	 * It ensures that required properties are set and that they are valid.
+	 *
+	 * @return WP_Error
+	 */
+	public function validate(): WP_Error {
+		$error = new WP_Error();
+		$error = $this->validate_properties( $error );
+
+		if ( $error->has_errors() ) {
+			return $error;
+		}
+
+		$error = $this->validate_values( $error );
+
+		return $error;
+	}
+
+	/**
+	 * Convert product to array
+	 *
+	 * If product is valid it will be transformed to array that can be sent to Omnisend.
+	 *
+	 * @return array
+	 */
+	public function to_array(): array {
+		if ( $this->validate()->has_errors() ) {
+			return array();
+		}
+
+		$arr = array(
+			'currency' => $this->currency,
+			'id'       => $this->id,
+			'status'   => $this->status,
+			'title'    => $this->title,
+			'url'      => $this->url,
+		);
+
+		if ( ! empty( $this->category_ids ) ) {
+			$arr['categoryIDs'] = $this->category_ids;
+		}
+
+		if ( ! empty( $this->created_at ) ) {
+			$arr['createdAt'] = $this->created_at;
+		}
+
+		if ( ! empty( $this->default_image_url ) ) {
+			$arr['defaultImageUrl'] = $this->default_image_url;
+		}
+
+		if ( ! empty( $this->description ) ) {
+			$arr['description'] = $this->description;
+		}
+
+		if ( ! empty( $this->images ) ) {
+			$arr['images'] = $this->images;
+		}
+
+		if ( ! empty( $this->tags ) ) {
+			$arr['tags'] = $this->tags;
+		}
+
+		if ( ! empty( $this->type ) ) {
+			$arr['type'] = $this->type;
+		}
+
+		if ( ! empty( $this->updated_at ) ) {
+			$arr['updatedAt'] = $this->updated_at;
+		}
+
+		if ( ! empty( $this->vendor ) ) {
+			$arr['vendor'] = $this->vendor;
+		}
+
+		if ( ! empty( $this->variants ) ) {
+			$arr['variants'] = $this->variants;
```

Comment:
shouldn't you use `productVariant` method `to_array` here? Also, product validation should validate variants and reuse `productVariant` validation, because I think you will be sending and creating products and variant is a child of a product and you dont want to call separate methods and deal with it

Or perhaps I am missing something here? How are you planing to use them from client?